x-pack/plugins/ingest_manager/public/applications/ingest_manager/components/package_icon.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useEffect, useMemo, useState } from 'react';
+import { ICON_TYPES, EuiIcon, EuiIconProps } from '@elastic/eui';
+import { PackageInfo, PackageListItem } from '../../../../common/types/models';
+import { useLinks } from '../sections/epm/hooks';
+import { epmRouteService } from '../../../../common/services';
+import { sendRequest } from '../hooks/use_request';
+import { GetInfoResponse } from '../types';
+type Package = PackageInfo | PackageListItem;
+
+const CACHED_ICONS = new Map<string, string>();
+
+export const PackageIcon: React.FunctionComponent<{
```

Comment:
Should `/epm/components/package_icon.tsx` be deleted and usages replaced with this one?

Reply:
Yes, I was waiting for a review before making the changes. I will do that refactoring after merging this PR :)